### ISSUE TITLE
Investigate the implementation of `aria-controls` on components using the animate library

### ISSUE THREAD

**Issue by electronicwoft** [govau/design-system-components#797]
Hi team. In my view, there are some WCAG/accessibility issues outstanding with the animation component. It is not possible to programmatically determine the current state of this widget (i.e., whether the animated element is visible) (4.1.2). Also, the aria-controls attribute on each control (open, close, toggle) points to content that may not be either visible or focusable. Using this widget with a screen reader will likely require unnecessary exploration to determine whether anything has taken place when any of the buttons are activated. this could be resolved if there was one button whose accessible name toggled in line with the visibility of the animated content and aria-controls was only present when the content was visible.

**Comment by sukhrajghuman** [comment]
hey @electronicwoft, this is **purely** a javascript component. The buttons/examples on our [doc site](https://designsystem.gov.au/components/animate/#default) you see are just for a visual test. This component is not to be used in isolation.

We use this component/javascript library to open and close our accordion, side-nav and main nav components.

If there are any issues with our implementations of these components from an accessibility point of view, please let us know.

**Comment by electronicwoft** [comment]
Hi @sukhrajghuman ....  yeah, sorry, could have worded the issue better. The aria-controls is present and points to something in the DOM even though whatever it is is not visible or focusable. It shouldn't do that until the element being controlled is visible and/or focusable. This can be the case for ANY specific implementation such as an accordian or side-nav. Also, it is arguable that aria-controls is unnecessary when controlling and controlled elements can be focused sequentially - see attached examples (please excuse the choppy 
[examples.zip](https://github.com/govau/design-system-components/files/3245948/examples.zip)
coding!).

**Comment by sukhrajghuman** [comment]
Yes I agree that we could remove the `aria-controls` in the [ animate example](https://designsystem.gov.au/components/animate/#default). I think the reason it was included in their was to demonstrate how to link buttons with sections/divs in the page. But it probably doesn't need to be there.

There is nothing we can really change in the `animate` component. This component has 1 job. To open and close things horizontally or vertically. 

We followed the example as recommended by the w3, our a11y team and several other sources that follow this pattern for accordions (using `aria-controls` even when content is not visible). If you believe this implementation should be changed, I will rename the title of this ticket  and open it up for the accessibility team and community for discussion.

Thoughts?

**Comment by electronicwoft** [comment]
Hi @sukhrajghuman. By all means, open this up for discussion.

The W3 examples are themselves contested. In my view, it is wise to use the ARIA 1.1 or 1.2 examples as a starting point and then test exhaustively.

I chose the animate component purely to save time thinking that the issue could then be extrapolated ... I can log issues for all similarly expanding components if that is required/desired. 

Setting aside the specific component for a moment, this issue is about the technique of using aria-controls for controlled elements that are not visible or focusable wherever it occurs. 

This issue may have been the result of #396. 

I am not against aria-controls per se just the use of aria-controls with a non-empty attribute value for content that is currently not visible/focusable. In my mind, it should be used sparingly for when moving focus is itself laborious such as having to tab two dozen times to get to a controlled element and not for items that are nearby in the DOM.

Also, it is not very well supported so there are many users who are missing out on the ability to move focus to a non-contiguous DOM node that, if considered to be desirable from an accessibility perspective, could be implemented using other mechanisms such as JavaScript.

**Comment by sukhrajghuman** [comment]
okay cool. I will do that :)

**Comment by electronicwoft** [comment]
Hi @sukhrajghuman  ... FYI, the latest update for JAWS changes its support for aria-controls:
"If you encounter an element on a web page with a defined ARIA controls relationship, JAWS will no longer say "use JAWSKEY+ALT+M to move to controlled element" by default. In most cases, the target of the controls relationship is adjacent to the element or does not provide any useful information."
[](https://support.freedomscientific.com/downloads/jaws/JAWSWhatsNew)
